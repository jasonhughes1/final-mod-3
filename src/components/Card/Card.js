import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ house }) => {
  console.log(house);

  const seats = house.seats.map( seat => <p>{seat}</p>);
  const titles = house.titles.map( title => <p>{title}</p>);
  const ancestralWeapons = house.ancestralWeapons.map( weapon => <p>{weapon}</p>);

  return (
    <div className='card'>
      <h1>{house.name}</h1>
      <p>{house.founded}</p>
      <h2>Seats</h2>
      <h3>{seats}</h3>
      <h2>Titles</h2>
      <h3>{titles}</h3>
      <h2>Coat Of Arms</h2>
      <p>{house.coatOfArms}</p>
      <h2>Ancestral Weapons</h2>
      <div>{ancestralWeapons}</div>
      <h3>{house.words}</h3>
    </div>
  );
};


export default Card;
