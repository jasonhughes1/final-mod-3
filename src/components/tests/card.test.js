import React from 'react';
import Card from '../Card/Card';
import { shallow } from 'enzyme';
import houseMock from './mockData/houseMock';

describe('Card Tests', () => {

  it('should render component and match snapshot', () => {
    const renderedCard = shallow(
      <Card house={houseMock[0]} />);

    expect(renderedCard).toMatchSnapshot();  
  });
});
