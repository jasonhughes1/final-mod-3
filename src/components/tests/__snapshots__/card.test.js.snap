// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tests should render component and match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    house={
        Object {
            "ancestralWeapons": Array [
              "Lady Forlorn",
            ],
            "coatOfArms": "Three black...",
            "diedOut": "",
            "founded": "",
            "name": "House Corbray of Heart's Home",
            "seats": Array [
              "Heart's Home",
            ],
            "titles": Array [
              "Lord of Heart's Home",
              "King of the Fingers (historical)",
            ],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          House Corbray of Heart's Home
</h1>,
        <p />,
        <h2>
          Seats
</h2>,
        <h3>
          <p>
                    Heart's Home
          </p>
</h3>,
        <h2>
          Titles
</h2>,
        <h3>
          <p>
                    Lord of Heart's Home
          </p>
          <p>
                    King of the Fingers (historical)
          </p>
</h3>,
        <h2>
          Coat Of Arms
</h2>,
        <p>
          Three black...
</p>,
        <h2>
          Ancestral Weapons
</h2>,
        <div>
          <p>
                    Lady Forlorn
          </p>
</div>,
        <h3 />,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "House Corbray of Heart's Home",
        },
        "ref": null,
        "rendered": "House Corbray of Heart's Home",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Seats",
        },
        "ref": null,
        "rendered": "Seats",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Heart's Home
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Heart's Home",
            },
            "ref": null,
            "rendered": "Heart's Home",
            "type": "p",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Titles",
        },
        "ref": null,
        "rendered": "Titles",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Lord of Heart's Home
</p>,
            <p>
              King of the Fingers (historical)
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lord of Heart's Home",
            },
            "ref": null,
            "rendered": "Lord of Heart's Home",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "King of the Fingers (historical)",
            },
            "ref": null,
            "rendered": "King of the Fingers (historical)",
            "type": "p",
          },
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Coat Of Arms",
        },
        "ref": null,
        "rendered": "Coat Of Arms",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Three black...",
        },
        "ref": null,
        "rendered": "Three black...",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Ancestral Weapons",
        },
        "ref": null,
        "rendered": "Ancestral Weapons",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Lady Forlorn
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lady Forlorn",
            },
            "ref": null,
            "rendered": "Lady Forlorn",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            House Corbray of Heart's Home
</h1>,
          <p />,
          <h2>
            Seats
</h2>,
          <h3>
            <p>
                        Heart's Home
            </p>
</h3>,
          <h2>
            Titles
</h2>,
          <h3>
            <p>
                        Lord of Heart's Home
            </p>
            <p>
                        King of the Fingers (historical)
            </p>
</h3>,
          <h2>
            Coat Of Arms
</h2>,
          <p>
            Three black...
</p>,
          <h2>
            Ancestral Weapons
</h2>,
          <div>
            <p>
                        Lady Forlorn
            </p>
</div>,
          <h3 />,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "House Corbray of Heart's Home",
          },
          "ref": null,
          "rendered": "House Corbray of Heart's Home",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
          },
          "ref": null,
          "rendered": "",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Seats",
          },
          "ref": null,
          "rendered": "Seats",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Heart's Home
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Heart's Home",
              },
              "ref": null,
              "rendered": "Heart's Home",
              "type": "p",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Titles",
          },
          "ref": null,
          "rendered": "Titles",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Lord of Heart's Home
</p>,
              <p>
                King of the Fingers (historical)
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lord of Heart's Home",
              },
              "ref": null,
              "rendered": "Lord of Heart's Home",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "King of the Fingers (historical)",
              },
              "ref": null,
              "rendered": "King of the Fingers (historical)",
              "type": "p",
            },
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Coat Of Arms",
          },
          "ref": null,
          "rendered": "Coat Of Arms",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Three black...",
          },
          "ref": null,
          "rendered": "Three black...",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ancestral Weapons",
          },
          "ref": null,
          "rendered": "Ancestral Weapons",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                Lady Forlorn
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lady Forlorn",
              },
              "ref": null,
              "rendered": "Lady Forlorn",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
